src/platform/nodejs/lib/util/crypto.ts

```diff
@@ -223,51 +214,58 @@ var createCryptoClass = function (bufferUtils: typeof BufferUtils) {
     algorithm: string;
     key: NodeCipherKey;
     iv: Buffer | null;
-    encryptCipher: NodeCipher;
-    blockLength: number;
+    encryptCipher: NodeCipher | null = null;
 
-    constructor(params: CipherParams, iv: Buffer) {
+    constructor(params: CipherParams, iv: Buffer | null) {
       this.algorithm = params.algorithm + '-' + String(params.keyLength) + '-' + params.mode;
       this.key = params.key;
       this.iv = iv;
-      this.encryptCipher = crypto.createCipheriv(this.algorithm, this.key, this.iv);
-      this.blockLength = this.iv.length;
     }
 
-    encrypt(plaintext: InputPlaintext, callback: (error: Error | null, data: OutputCiphertext | null) => void) {
+    async encrypt(plaintext: InputPlaintext): Promise<OutputCiphertext> {
       Logger.logAction(Logger.LOG_MICRO, 'CBCCipher.encrypt()', '');
+
+      const iv = await this.getIv();
+      if (!this.encryptCipher) {
+        this.encryptCipher = crypto.createCipheriv(this.algorithm, this.key, iv);
```

Comment:
If I understand it correctly, `this.encryptCipher` is checked and `crypto.createCipheriv` is called here again (in additiona to `getIv`) because we may have provided iv value for CBCCipher constructor, so the first call to `getIv` will enter the first if and won't create encryptCipher.
I suggest moving this also to `getIv` method, inside `if (this.iv) {` statement to cover this specific case.
For future-proofing.

Reply:
Yep, your understanding of the motivation is correct. You're right, it doesn't really make sense to spread the creation of `encryptCipher` across two separate places. But I think that inside `encrypt()` is the more appropriate place for it (so that method can be sure that `encryptCipher` is always non-null when it comes to use it, without having to understand the side effects of `getIv()`), so I've moved it there.

Reply:
Agree! Thank you